File: app/src/main/java/com/bkjcb/rqapplication/base/util/RxJavaUtil.java
package com.bkjcb.rqapplication.base.util;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by DengShuai on 2020/5/28.
 * Description :
 */
public class RxJavaUtil {
    public static <UD> ObservableTransformer<UD, UD> getObservableTransformer() {
        return new ObservableTransformer<UD, UD>() {
            @Override
            public ObservableSource<UD> apply(Observable<UD> upstream) {
                return upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
            }
        };
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/base/util/RxJavaUtil.kt
package com.bkjcb.rqapplication.base.util

import io.reactivex.ObservableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Created by DengShuai on 2020/5/28.
 * Description :
 */
object RxJavaUtil {
    fun <UD> getObservableTransformer(): ObservableTransformer<UD, UD> {
        return ObservableTransformer { upstream -> upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()) }
    }
}